### PR TITLE
[BUGFIX] assorted thought bug fixes

### LINKED ISSUE #5935: [BUG] thought generator is not properly generating thoughts
### Pre-requisites

- [x] I have searched the existing issues
- [x] This issue was encountered (or replicated) on a non-edited Clan
- [x] I confirm that I am NOT playing on a modded version of the game (e.g. LifeGen)

### Description

When using the thought generator option in the game config it is only selecting the first applicable cat to have the thought instead of all applicable cats and claiming that other cats do not fit the constraints.

I tested this using the gen_clan constraint which is applicable to every single clancat. 

code entered:
Image

functioning leader:
Image

non functioning cats:
Image
Image

error in console:
Image

### Grade

Non-vital - Can easily be ignored and does not have a large impact on play

### Gamemode

Expanded Mode

### Challenge cards

_No response_

### Steps to reproduce the behavior

1. use the thought generator to generate "gen_clan"
2. it will likely appear on the leader and no other cats. 

### Patrol ID + Outcome Print (if applicable)

_No response_

### Commit # or Game Version Number

d4a98e472e7803c9d3222d4832bdb57f92db2b1a

### Error log

_No response_

### Additional context

_No response_

### Screenshots

_No response_

### LINKED ISSUE #5945: [BUG] Death Thought Stuck at First Thought
### Pre-requisites

- [x] I have searched the existing issues
- [x] This issue was encountered (or replicated) on a non-edited Clan
- [x] I confirm that I am NOT playing on a modded version of the game (e.g. LifeGen)

### Description

Dead cats' statuses are just stuck on the one they had since dying

### Grade

Regular - Should be handled soon, but doesn't necessarily break the game or inhibit the player

### Gamemode

Cruel Season

### Challenge cards

Survival of fittest
Kit Season
Mouths to Feed
Forsaken
For the Throne
Lean Meat
Always Leaf Fall
Public Enemy
Meagre stocks

### Steps to reproduce the behavior

Kill cats I guess???????

### Patrol ID + Outcome Print (if applicable)

_No response_

### Commit # or Game Version Number

b11a89ec

### Error log

_No response_

### Additional context

It isn't cruel clan exclusive, when a friend and I encountered this bug vi went ahead and tested it on a fresh save of expanded too and it happened there too

EclipseClan is my friend's save, GrinningClan is mine, and PartridgeClan is the test clan
This might be tied to #5939 but doesn't entirely seem to be the case? We haven't had the game break on us yet just ain't changing the thoughts

### Screenshots

Image

Image

Image

### LINKED ISSUE #5980: [BUG] lost cats dont have lost thoughts
### Pre-requisites

- [x] I have searched the existing issues
- [x] This issue was encountered (or replicated) on a non-edited Clan
- [x] I confirm that I am NOT playing on a modded version of the game (e.g. LifeGen)

### Description

I was adding lost thoughts to my branches and when I went to test it, I noticed not only did my lost thoughts not pop up but none of them showed even though I am completely certain they should work

### Grade

Non-vital - Can easily be ignored and does not have a large impact on play

### Gamemode

Classic Mode

### Challenge cards

_No response_

### Steps to reproduce the behavior

Force event where cats gets lost
Check thoughts
Repeat and notice that none of the thoughts from while_alive\while_lost\general.json appears

### Patrol ID + Outcome Print (if applicable)

_No response_

### Commit # or Game Version Number

44bfe42c

### Error log

_No response_

### Additional context

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by orbitaclass**
## About The Pull Request

- Fixes ensured thoughts only generating for the first applicable cat; this was because the ensured thought's id was being added to `used_thoughts`, preventing it from being used again.
- Fixes dead cats not being assigned new thoughts on moonskip.
- Adds `while_lost/general.json` to the pool of allowed thoughts for lost cats.
- Removes unnecessary status constraints from `while_lost/general.json`.
- `leave_clan` now assigns thoughts from `on_rank_change`. This will utilize a few thoughts that were previously unused.
- Edits constraints in `while_alive/rogue.json` to make more sense when generated.

## Linked Issues
Fixes: #5945
Fixes: #5935
Fixes: #5980

## Proof of Testing

Screenshot 2026-09-18 204345\
Screenshot 2026-09-18 204529

cat other than leader with ensured thought

Screenshot 2026-09-18 203650\
Screenshot 2026-09-18 203733\
dead cat's thought changes on moonskip

Screenshot 2026-09-18 203055\
thought from `while_lost/general.json`

Screenshot 2026-09-18 204019

thought from `on_rank_change/kittypet.json `(previously unused)

### CODE REVIEW

File: resources/lang/en/thoughts/while_alive/while_lost/general.json
Comment (scribblecrumb) [inline resources/lang/en/thoughts/while_alive/while_lost/general.json:177]: While we're fixing things, this also needs an `r_c` entry in `involved_cats`, since it references `r_c` in the relationship constraint!

File: scripts/events.py [hunk 1]
```diff
@@ -996,6 +996,7 @@ def one_moon_cat(cat):
             cat.moons += 1
         else:
             cat.status.increase_current_moons_as()
+        cat.assign_thought()
```
Comment (scribblecrumb) [inline scripts/events.py:999]: Were dead cats not generating a thought upon opening their profile? Cats shouldn't generate thoughts on moonskip, we recently changed them to generate upon profile opening as a way to prevent lag.

**Review by scribblecrumb — CHANGES_REQUESTED** [review]

File: scripts/events.py (hunk 1 above)
Comment (orbitaclass) [reply scripts/events.py:999]: Dead cats aren't generating thoughts upon opening their profile because `build_profile` checks that there isn't an existing thought before generating a new one. I noticed that living cats' thoughts are reset to None when `assign_thought` is called in `one_moon`, and `build_profile` does generate a new thought in that case. For dead cats, it doesn't, since one_moon_cat returns before it calls one_moon and their thoughts aren't reset. When I tried removing `assign_thought` from `one_moon`, living cats didn't generate any new thoughts, which seems to me like it being called to reset thoughts is currently needed for `build_profile` to generate a thought in the first place.

Comment (scribblecrumb) [reply scripts/events.py:999]: Oooh you know what, that's on me for forgetting what `assign_thought()` does lmao. I'm even the one who wrote that code, geez.

**Review by scribblecrumb — APPROVED** [review]
